### PR TITLE
Components

### PR DESCRIPTION
**Pull request by marcelomaidden**
## Components
At this pull request, I add Redux functionalities and connect the App to the Redux store:
- Changes project folder structure adding components, containers, actions and, reducer folders
- Adds Book, BooksForm, BookList components
- Adds CREATE_BOOK, REMOVE_BOOK actions
- Uses combineReducers to create the root reducer
- Creates the store with the default state for the application
- Adds Bootstrap and basic CSS for form and table

### CODE REVIEW

File: src/containers/BookList.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+BookList.propTypes = {
+  books: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.number.isRequired,
+      title: PropTypes.string.isRequired,
+      category: PropTypes.string.isRequired,
+    }),
+  ).isRequired,
+};
+
```
Comment (addod19) [inline src/containers/BookList.js:22]: - Our BookList Component should connect to Redux store and fetches books from the state
-  It also uses `mapStateToProps` to fetch books from the `store`
- It displays books list in HTML table using `Book` component

File: src/components/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  { key: 0, name: 'Action' },
+  { key: 1, name: 'Biography' },
+  { key: 2, name: 'History' },
+  { key: 3, name: 'Horror' },
+  { key: 4, name: 'Kids' },
+  { key: 5, name: 'Learning' },
+  { key: 6, name: 'Sci-Fi' },
+];
+
+const BooksForm = () => (
```
Comment (addod19) [inline src/components/BooksForm.js]: - Our BookForm component must be in the src/containers directory

File: src/containers/App.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import BookList from './BookList';
+import BooksForm from '../components/BooksForm';
+import { createBook, deleteBook } from '../actions';
+
+const App = ({ books }) => (
```
Comment (addod19) [inline src/containers/App.js]: Our App component must be in the src/components directory! Let us fix it :smile:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @marcelomaidden, great job on this project :+1: The PR is professional with a title and a short description :slightly_smiling_face: The README file is professional :fireworks: Commit messages well written :100: However, I have let a comment in some of the files we need to fix. Kindly address them :+1: 

Once these are fixed, kindly submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by SpaYco — APPROVED** [review]
# Project Approved

thank you for making the required changes, your project is now approved!

### please don't forget to merge to master, good luck with the next projects!


Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Pull request merged**